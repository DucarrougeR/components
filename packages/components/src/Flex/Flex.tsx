import React, { FunctionComponent, Ref } from 'react'
import styled, { css, StyledComponent } from 'styled-components'
import { Box, BoxBaseProps, BoxFlexProps } from '../Box'

/**
 * styled-system has its own FlexProps, so we call this one FlexBoxProps to disambiguate.
 */
export interface FlexProps
  extends Omit<BoxBaseProps<HTMLDivElement>, 'display' | 'as'>,
    BoxFlexProps {
  hidden?: boolean
}

type ComponentType = FunctionComponent<FlexProps>
type StyledComponentType = StyledComponent<ComponentType, FlexProps>

const InternalFlex: ComponentType = props => {
  return (
    <Box display="flex" {...props}>
      {props.children}
    </Box>
  )
}

const FlexFactory = React.forwardRef<StyledComponentType, FlexProps>(
  (props: FlexProps, ref: Ref<StyledComponentType>) => (
    <InternalFlex ref={ref} {...props} />
  )
)

/** @component */
export const Flex = styled<ComponentType>(FlexFactory)`
  ${(props: FlexProps) => hidden(props.hidden)};
`

const hidden = (hide?: boolean) =>
  hide
    ? css`
        display: none;
      `
    : false
