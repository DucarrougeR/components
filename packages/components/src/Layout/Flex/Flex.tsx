import { CompatibleHTMLProps, FlexboxProps } from '@looker/design-tokens'
import styled from 'styled-components'
import { LayoutComponentProps, layoutCSS } from '../layout'

/**
 * styled-system has its own FlexBoxProps, so we call this one FlexProps to disambiguate.
 */
export interface FlexProps
  extends CompatibleHTMLProps<HTMLDivElement>,
    Omit<LayoutComponentProps, 'display'>,
    FlexboxProps {}

/** @component */
export const Flex = styled.div<FlexProps>`
  ${layoutCSS}
  display: flex;
`
