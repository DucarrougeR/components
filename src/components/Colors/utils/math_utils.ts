/**
 * Utility to convert a numerical value to a percent given the value and a max value.
 */
export const toPerc = (value: number, maxValue: number) =>
  Math.round((value / maxValue) * 100)
