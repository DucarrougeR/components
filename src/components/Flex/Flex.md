```js noeditor
const StatusAndResources = require('../../../styleguide_components/StatusAndResources').StatusAndResources;
<StatusAndResources
  status="experimental"
  githubURL="https://github.com/looker/lens/blob/master/src/components/Flex/Flex.tsx"
  feedbackTitle="Flex Component Feedback" />
```

The `<Flex />` component creates a flex container for you to place `<FlexItem />` components inside. It lets you utilize [flex box](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout) to create a wide variety of layouts, with powerful alignment, directional, and space distribution abilities.

If you are not familiar with flex box, [here is a great guide](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) for how it works.

### Basic Usage
The `<Flex />` component has the ability to control the layout direction and order of its children. By default the children will layout in a row.

```js
<Flex>
  <FlexItem>👋</FlexItem>
  <FlexItem>💪</FlexItem>
  <FlexItem>📦</FlexItem>
</Flex>
```

### A real world example
Using our `<Flex>` and `<FlexItem>` components we can quickly scaffold out real world use cases like a navigation bar.
```js
<Flex bg="palette.charcoal800" px="small" height="50px" alignItems="center" justifyContent="space-between">
  <FlexItem mr="small" pt="xsmall"><IconHamburger fill="white" width="24" height="24" /></FlexItem>
  <FlexItem><IconLookerLogo fill="white" width="60px" height="auto"/></FlexItem>
  <FlexItem flex="1">
    <Flex justifyContent="center">
    <Box px="small"><Text size="5" style={{'color': '#fff'}}>Browse <IconExpandMore fill="white" /></Text></Box>
    <Box px="small"><Text size="5" style={{'color': '#fff'}}>Explore <IconExpandMore fill="white" /></Text></Box>
    <Box px="small"><Text size="5" style={{'color': '#fff'}}>Develop <IconExpandMore fill="white" /></Text></Box>
    <Box px="small"><Text size="5" style={{'color': '#fff'}}>Admin <IconExpandMore fill="white" /></Text></Box>
    </Flex>
  </FlexItem>
  <FlexItem>
    <Flex>
      <FlexItem mr="small"><IconSearch fill="white" width="24" height="24" /></FlexItem>
      <FlexItem mr="small"><IconLiveHelp fill="white" width="24" height="24" /></FlexItem>
      <FlexItem><IconGroup fill="white" width="24" height="24" /></FlexItem>
    </Flex>
  </FlexItem>
</Flex>
```


### Changing direction and order

You can change the layout direction and order of flex children by using the `flexDirection` property. Below we use the same children in each example but can adjust the layout quickly using `flexDirection`.

```js
<Flex>
  <Flex mr="large">
    <FlexItem>1️⃣</FlexItem>
    <FlexItem>2️⃣</FlexItem>
    <FlexItem>3️⃣</FlexItem>
  </Flex>

  <Flex flexDirection="column" mr="large">
    <FlexItem>1️⃣</FlexItem>
    <FlexItem>2️⃣</FlexItem>
    <FlexItem>3️⃣</FlexItem>
  </Flex>

  <Flex flexDirection="column-reverse" mr="large">
    <FlexItem>1️⃣</FlexItem>
    <FlexItem>2️⃣</FlexItem>
    <FlexItem>3️⃣</FlexItem>
  </Flex>

  <Flex flexDirection="row-reverse" mr="large">
    <FlexItem>1️⃣</FlexItem>
    <FlexItem>2️⃣</FlexItem>
    <FlexItem>3️⃣</FlexItem>
  </Flex>
</Flex>
```

### Item distribution

The `<Flex>` component allows you to control how its children are distributed within its container. Using the `justifyContent` property you can adjust the distribution of the children. Below is a table with all the justify options:
```js noeditor
<Table>
  <TableHead>
    <TableRow>
      <TableHeaderCell width="40%">Justify Property</TableHeaderCell>
      <TableHeaderCell>Behavior</TableHeaderCell>
    </TableRow>
  </TableHead>
  <TableBody>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>center</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Pack items around the center</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>flex-start</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Pack items from the start</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>flex-end</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Pack items from the end</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>space-between</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Distribute items evenly. The first item is flush with the start, the last is flush with the end </Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>space-around</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Distribute items evenly. Items have a half-size space on either end</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>space-evenly</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Distribute items evenly. Items have equal space around them</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>stretch</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Distribute items evenly. Items strechted to fit the container</Text></TableDataCell>
    </TableRow>
  </TableBody>
</Table>
```

Here is a visual example of how using `justifyContent` affects the distribution of items in a `<Flex />` component.

```js
<div>

<Flex justifyContent="flex-start" mb="medium" bg="palette.charcoal000">
  <Box><Text size="xxxxlarge">🕺</Text></Box>
  <Box><Text size="xxxxlarge">🎶</Text></Box>
  <Box><Text size="xxxxlarge">💃</Text></Box>
</Flex>

<Flex justifyContent="space-between"  mb="medium" bg="palette.charcoal000">
  <Box><Text size="xxxxlarge">🕺</Text></Box>
  <Box><Text size="xxxxlarge">🎶</Text></Box>
  <Box><Text size="xxxxlarge">💃</Text></Box>
</Flex>

<Flex justifyContent="space-around"  mb="medium" bg="palette.charcoal000">
  <Box><Text size="xxxxlarge">🕺</Text></Box>
  <Box><Text size="xxxxlarge">🎶</Text></Box>
  <Box><Text size="xxxxlarge">💃</Text></Box>
</Flex>

<Flex justifyContent="space-evenly"  mb="medium" bg="palette.charcoal000">
  <Box><Text size="xxxxlarge">🕺</Text></Box>
  <Box><Text size="xxxxlarge">🎶</Text></Box>
  <Box><Text size="xxxxlarge">💃</Text></Box>
</Flex>

<Flex justifyContent="center"  mb="medium" bg="palette.charcoal000">
  <Box><Text size="xxxxlarge">🕺</Text></Box>
  <Box><Text size="xxxxlarge">🎶</Text></Box>
  <Box><Text size="xxxxlarge">💃</Text></Box>
</Flex>

<Flex justifyContent="flex-end"  mb="medium" bg="palette.charcoal000">
  <Box><Text size="xxxxlarge">🕺</Text></Box>
  <Box><Text size="xxxxlarge">🎶</Text></Box>
  <Box><Text size="xxxxlarge">💃</Text></Box>
</Flex>
</div>
```

### Aligning Items

Flex box allows for quickly adjusting the alignment of items inside the flex container. Below are available alignment types:

```js noeditor
<Table>
  <TableHead>
    <TableRow>
      <TableHeaderCell width="40%">AlignItems Property</TableHeaderCell>
      <TableHeaderCell>Behavior</TableHeaderCell>
    </TableRow>
  </TableHead>
  <TableBody>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>center</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Items are aligned center of container</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>flex-start</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Items are aligned to the start of the container</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>flex-end</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Items are aligned to the start of the container</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>baseline</Code></span></TableDataCell>
      <TableDataCell><Text size="5">All items are aligned so their baselines align</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>stretch</Code></span> <Span size="xsmall">(default)</Span></TableDataCell>
      <TableDataCell><Text size="5">Items strechted to fit the container</Text></TableDataCell>
    </TableRow>
  </TableBody>
</Table>
```

Here is an example of using `alignItems` to adjust how items are aligned within the flex container.
```js
<div>
<Flex mb="xxlarge">
  <FlexItem p="xxlarge" bg="palette.purple200">Default (stretch)</FlexItem>
  <FlexItem p="large" bg="palette.blue200">One</FlexItem>
  <FlexItem p="medium" bg="palette.green200">Two</FlexItem>
  <FlexItem p="small" bg="palette.red200">Three</FlexItem>
</Flex>

<Flex alignItems="center" mb="xxlarge">
  <FlexItem p="xxlarge" bg="palette.purple200">Center</FlexItem>
  <FlexItem p="large" bg="palette.blue200">One</FlexItem>
  <FlexItem p="medium" bg="palette.green200">Two</FlexItem>
  <FlexItem p="small" bg="palette.red200">Three</FlexItem>
</Flex>

<Flex alignItems="flex-start" mb="xxlarge">
  <FlexItem p="xxlarge" bg="palette.purple200">Start</FlexItem>
  <FlexItem p="large" bg="palette.blue200">Two</FlexItem>
  <FlexItem p="medium" bg="palette.green200">Three</FlexItem>
  <FlexItem p="small" bg="palette.red200">Four</FlexItem>
</Flex>

<Flex alignItems="flex-end" mb="xxlarge">
  <FlexItem p="xxlarge" bg="palette.purple200">End</FlexItem>
  <FlexItem p="large" bg="palette.blue200">Two</FlexItem>
  <FlexItem p="medium" bg="palette.green200">Three</FlexItem>
  <FlexItem p="small" bg="palette.red200">Four</FlexItem>
</Flex>

<Flex alignItems="baseline" mb="xxlarge">
  <FlexItem p="xxlarge" bg="palette.purple200">Baseline</FlexItem>
  <FlexItem p="large" bg="palette.blue200">Two</FlexItem>
  <FlexItem p="medium" bg="palette.green200">Three</FlexItem>
  <FlexItem p="small" bg="palette.red200">Four</FlexItem>
</Flex>
</div>
```

### Aligning Content
If you have multiple rows of flex items, the `alignContent` property determines how the rows are distributed within the flex container.

```js noeditor
<Table>
  <TableHead>
    <TableRow>
      <TableHeaderCell width="40%">AlignContent Property</TableHeaderCell>
      <TableHeaderCell>Behavior</TableHeaderCell>
    </TableRow>
  </TableHead>
  <TableBody>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>center</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Lines are packed to the center of container</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>flex-start</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Lines are packed to the start of the container</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>flex-end</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Lines are packed to the start of the container</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>space-between</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Lines evenly distributed. First line at start of container, last line at end of container</Text></TableDataCell>
    </TableRow>
        <TableRow>
      <TableDataCell><span class="prop-code"><Code>space-around</Code></span></TableDataCell>
      <TableDataCell><Text size="5">Lines evenly distributed. Even space around each line</Text></TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell><span class="prop-code"><Code>stretch</Code></span> <Span size="xsmall">(default)</Span></TableDataCell>
      <TableDataCell><Text size="5">Items strechted to fit the container</Text></TableDataCell>
    </TableRow>
  </TableBody>
</Table>
```

```js
<Flex justifyContent="space-between">
  <Flex alignContent="space-around" width="30%" height="150px" flexWrap="wrap" bg="palette.purple400">
    <FlexItem width="70%" m="small" p="xsmall" bg="palette.purple200">Aligned w/</FlexItem>
    <FlexItem width="30%" m="small" p="xsmall" bg="palette.purple200">Space</FlexItem>
    <FlexItem width="50%" m="small" p="xsmall" bg="palette.purple200">Betwen</FlexItem>
  </Flex>

  <Flex alignContent="flex-end" width="30%" height="150px" flexWrap="wrap" bg="palette.blue400">
    <FlexItem width="70%" m="small" p="xsmall" bg="palette.blue200">Aligned w/</FlexItem>
    <FlexItem width="30%" m="small" p="xsmall" bg="palette.blue200">Flex</FlexItem>
    <FlexItem width="50%" m="small" p="xsmall" bg="palette.blue200">End</FlexItem>
  </Flex>

  <Flex alignContent="center" width="30%" height="150px" flexWrap="wrap" bg="palette.green400">
    <FlexItem width="70%" m="small" p="xsmall" bg="palette.green200">Aligned w/</FlexItem>
    <FlexItem width="30%" m="small" p="xsmall" bg="palette.green200">Flex</FlexItem>
    <FlexItem width="50%" m="small" p="xsmall" bg="palette.green200">Center</FlexItem>
  </Flex>
</Flex>
```


### Wrapping Flex Items

The  `flexWrap` property determines if flex items should be forced into a single line or if they can wrap onto multiple lines.

```js
<Flex width="80%" flexWrap="nowrap" mb="large">
  <FlexItem width="40%" p="large" bg="palette.red200">These Lines</FlexItem>
  <FlexItem width="40%" p="large" bg="palette.blue200">Will NOT</FlexItem>
  <FlexItem width="40%" p="large" bg="palette.green200">Wrap</FlexItem>
</Flex>

<Flex width="50%" flexWrap="wrap" mb="large">
  <FlexItem width="40%" p="large" bg="palette.red200">These Lines</FlexItem>
  <FlexItem width="40%" p="large" bg="palette.blue200">Will</FlexItem>
  <FlexItem width="40%" p="large" bg="palette.green200">Wrap</FlexItem>
</Flex>

<Flex width="50%" flexWrap="wrap-reverse" mb="large">
  <FlexItem width="40%" p="large" bg="palette.red200">These Lines</FlexItem>
  <FlexItem width="40%" p="large" bg="palette.blue200">Will</FlexItem>
  <FlexItem width="40%" p="large" bg="palette.green200">Wrap Reverse</FlexItem>
</Flex>
```

### Responsive Behavior

Many of the properties you can set on the `<Flex />` component will accept an array of values that will then be applied at different breakpoints. There are 5 [breakpoints](/#!/Breakpoints) slots that you can target. Here is an example of what that looks like.

```js
<Flex
  flexDirection={[
      "column", // column up to the first breakpoint
      null,     // stay a column past first breakpoint
      "row"     // switch to row layout after second breakpoint
  ]}>

  <FlexItem p="medium" bg="palette.yellow200">👋</FlexItem>
  <FlexItem p="medium" bg="palette.green200">💪</FlexItem>
  <FlexItem p="medium" bg="palette.purple200">📦</FlexItem>
</Flex>
```
