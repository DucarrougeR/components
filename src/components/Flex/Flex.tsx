import * as React from 'react'
import { css, styled } from '../../style'
import { Box, BoxBasePropsWithout, BoxFlexProps } from '../Box'

/**
 * styled-system has its own FlexProps, so we call this one FlexBoxProps to disambiguate.
 */
export interface FlexProps
  extends BoxBasePropsWithout<HTMLDivElement, 'display'>,
    BoxFlexProps {
  hidden?: boolean
}

const InternalFlex: React.FC<FlexProps> = props => {
  return (
    <Box display="flex" {...props}>
      {props.children}
    </Box>
  )
}

const FlexFactory = React.forwardRef((props: FlexProps, ref) => (
  <InternalFlex innerRef={ref} {...props} />
))

export const Flex = styled(FlexFactory)`
  ${props => hidden(props.hidden)};
`

const hidden = (hide?: boolean) =>
  hide
    ? css`
        display: none;
      `
    : false
