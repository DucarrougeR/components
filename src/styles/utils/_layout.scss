// Returns a value from the $spacing list
@function spacing($key) {
  @return map-lookup($spacing, $key, '$spacing');
}
