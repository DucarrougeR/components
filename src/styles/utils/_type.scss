// Returns a value from the $test-size list
@function font-size($key) {
  @return map-lookup($text-size, $key, '$text-size');
 }

 // Returns a value from the $line-height list
 @function line-height($key) {
   @return map-lookup($line-height, $key, '$line-height');
 }

 @mixin fontSize($size) {
  font-size: $size;
  font-size: pxToRem($size);
}
