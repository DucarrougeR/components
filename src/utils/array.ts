/**
 * Returns an array of length `size` filled with zeros.
 * Useful for looping constructs where a length is known, but no array is provided.
 */

export const sizedArray = (size: number) => Array(size).fill(0)
