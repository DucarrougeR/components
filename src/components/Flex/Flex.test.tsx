import 'jest-styled-components'
import * as React from 'react'
import { createWithTheme } from '../../../test/utils/create_with_theme'
import { assertSnapshot } from '../../../test/utils/snapshot'
import { Flex } from './Flex'

test('Flex default', () => {
  const component = createWithTheme(
    <Flex>
      <div>ü•ë</div>
      <div>üêõ</div>
    </Flex>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Flex supports justifyContent ', () => {
  assertSnapshot(<Flex justifyContent="center" />)
  assertSnapshot(<Flex justifyContent="start" />)
  assertSnapshot(<Flex justifyContent="end" />)
  assertSnapshot(<Flex justifyContent="space-between" />)
  assertSnapshot(<Flex justifyContent="space-around" />)
  assertSnapshot(<Flex justifyContent="space-evenly" />)
  assertSnapshot(<Flex justifyContent="stretch" />)
})

test('Flex supports alignItems ', () => {
  assertSnapshot(<Flex alignItems="center" />)
  assertSnapshot(<Flex alignItems="start" />)
  assertSnapshot(<Flex alignItems="end" />)
  assertSnapshot(<Flex alignItems="baseline" />)
  assertSnapshot(<Flex alignItems="stretch" />)
})

test('Flex supports alignContent ', () => {
  assertSnapshot(<Flex alignContent="center" />)
  assertSnapshot(<Flex alignContent="start" />)
  assertSnapshot(<Flex alignContent="end" />)
  assertSnapshot(<Flex alignContent="baseline" />)
  assertSnapshot(<Flex alignContent="space-between" />)
  assertSnapshot(<Flex alignContent="space-around" />)
  assertSnapshot(<Flex alignContent="space-evenly" />)
  assertSnapshot(<Flex alignContent="stretch" />)
})

test('Flex supports alignSelf ', () => {
  assertSnapshot(<Flex alignSelf="center" />)
  assertSnapshot(<Flex alignSelf="start" />)
  assertSnapshot(<Flex alignSelf="end" />)
  assertSnapshot(<Flex alignSelf="baseline" />)
  assertSnapshot(<Flex alignSelf="stretch" />)
})

test('Flex supports flexDirection ', () => {
  assertSnapshot(<Flex flexDirection="row" />)
  assertSnapshot(<Flex flexDirection="row-reverse" />)
  assertSnapshot(<Flex flexDirection="column" />)
  assertSnapshot(<Flex flexDirection="column-reverse" />)
})

test('Flex supports flexWrap ', () => {
  assertSnapshot(<Flex flexWrap="wrap" />)
  assertSnapshot(<Flex flexWrap="nowrap" />)
  assertSnapshot(<Flex flexWrap="wrap-reverse" />)
})
