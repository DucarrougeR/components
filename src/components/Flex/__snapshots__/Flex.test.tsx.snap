// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
>
  <div>
    ü•ë
  </div>
  <div>
    üêõ
  </div>
</div>
`;

exports[`Flex supports alignContent  1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

<div
  className="c0 c1"
/>
`;

exports[`Flex supports alignItems  1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0 c1"
/>
`;

exports[`Flex supports alignSelf  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
}

<div
  className="c0"
/>
`;

exports[`Flex supports flexDirection  1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

<div
  className="c0 c1"
/>
`;

exports[`Flex supports flexWrap  1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

<div
  className="c0 c1"
/>
`;

exports[`Flex supports justifyContent  1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0 c1"
/>
`;
